raphtory-graphql/src/model/graph/property.rs

```diff
@@ -1,25 +1,182 @@
 use dynamic_graphql::{ResolvedObject, ResolvedObjectFields};
-use raphtory::core::Prop;
+use raphtory::{
+    core::Prop,
+    db::api::properties::{
+        dyn_props::{DynConstProperties, DynProperties, DynProps, DynTemporalProperties},
+        TemporalPropertyView,
+    },
+};
 
 #[derive(ResolvedObject)]
-pub(crate) struct Property {
+pub(crate) struct GqlProp {
     key: String,
-    value: Prop,
+    prop: Prop,
+}
+impl GqlProp {
+    pub(crate) fn new(key: String, prop: Prop) -> Self {
+        Self { key, prop }
+    }
+}
+impl From<(String, Prop)> for GqlProp {
+    fn from(value: (String, Prop)) -> Self {
+        GqlProp::new(value.0, value.1)
+    }
 }
 
-impl Property {
-    pub(crate) fn new(key: String, value: Prop) -> Self {
-        Self { key, value }
+#[ResolvedObjectFields]
+impl GqlProp {
+    async fn key(&self) -> String {
+        self.key.clone()
+    }
+    async fn as_string(&self) -> String {
+        self.prop.to_string()
+    }
+}
+
+#[derive(ResolvedObject)]
+pub(crate) struct GqlTemporalProp {
+    key: String,
+    prop: TemporalPropertyView<DynProps>,
+}
+impl GqlTemporalProp {
+    pub(crate) fn new(key: String, prop: TemporalPropertyView<DynProps>) -> Self {
+        Self { key, prop }
+    }
+}
+impl From<(String, TemporalPropertyView<DynProps>)> for GqlTemporalProp {
+    fn from(value: (String, TemporalPropertyView<DynProps>)) -> Self {
+        GqlTemporalProp::new(value.0, value.1)
     }
 }
 
 #[ResolvedObjectFields]
-impl Property {
+impl GqlTemporalProp {
     async fn key(&self) -> String {
-        self.key.to_string()
+        self.key.clone()
+    }
+    async fn history(&self) -> Vec<i64> {
+        self.prop.history()
+    }
+    async fn values(&self) -> Vec<String> {
+        self.prop.values().iter().map(|x| x.to_string()).collect()
+    }
+    async fn at(&self, t: i64) -> Option<String> {
+        self.prop.at(t).map(|x| x.to_string())
+    }
+    async fn latest(&self) -> Option<String> {
+        self.prop.latest().map(|x| x.to_string())
+    }
+}
+
+#[derive(ResolvedObject)]
+pub(crate) struct GqlProperties {
+    props: DynProperties,
+}
+impl GqlProperties {
+    pub(crate) fn new(props: DynProperties) -> Self {
+        Self { props }
+    }
+}
+
+impl From<DynProperties> for GqlProperties {
```

Comment:
Make this `impl<P: Into<DynProperties>> From<P>` and a lot of weird conversions go away.